src/common/vector_operations/comparison_operators.cpp

```diff
@@ -190,12 +194,19 @@ static void NestedComparisonExecutor(Vector &left, Vector &right, Vector &result
 	for (idx_t i = 0; i < match_count; ++i) {
 		const auto idx = true_sel.get_index(i);
 		result_data[idx] = true;
+		// if the row was valid during the null check, set it to valid here as well
+		if (copy.RowIsValid(idx)) {
+			result_validity.SetValid(idx);
+		}
```

Comment:
Just a clarification question: `copy` contains the `result_validity` before `ComparisonSelector::Select`, which possibly overwriters some fields to invalid. Thus, can we reset them back to valid?

Nit: Maybe this will be clearer if we rename `copy` to something like `original_mask`?

Reply:
EDIT: I just saw your updated PR description. Makes sense to me. 👍 Should we rename `copy` to `original_mask` or `null_comparison_result`, or something similar?